### PR TITLE
POM and modules

### PR DESCRIPTION
**Pull request by MicmladenTo**
JSON files largely replaced with POM getters.
Functions extracted into modules to avoid repeating.

Pending:
Extract response validations into functions.

### CODE REVIEW

File: cypress/models/boardModule.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		this.nextButton.click();
+		this.nextButton.click();
+		
+		// Assert a successful post request and verify board name and URL from the response
+		cy.wait('@postABoard')
+		.its('response')
+		.then((res) => {
+		expect(res.statusCode).to.eq(201);
+		expect(res.body.name).to.eq(boardTitle);
+		cy.url().should('eq', Cypress.config().baseUrl + 'boards/' + res.body.id);
```
Comment (bdimitrovski) [inline cypress/models/boardModule.js]: Please, for the love of God, try writing nice Javascript

```suggestion
		cy.url().should('eq', `${Cypress.config().baseUrl}boards/${res.body.id}`);
```

File: cypress/models/organisationModule.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		this.navigateNextButton.click();
+
+		this.confirmBoardsModal.click();
+
+		cy.wait('@addOrganization')
+		.its('response')
+		.then((res) => {
+		expect(res.statusCode).to.eq(200);
+		expect(res.body.name).to.eq(organisationName);
+		cy.url().should('eq', Cypress.config().baseUrl + 'organizations/' + res.body.id + '/boards');
```
Comment (bdimitrovski) [inline cypress/models/organisationModule.js]: Nice JS much? 😄

File: cypress/models/organisationModule.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		})
+	},
+
+	editOrgName({organisationName = data.accountDetails.organisationName}) {
+		cy.intercept('PUT', '/api/v2/organizations/*').as('rename');
+
+		this.organisationNameHeader.eq(0).click();
+		if (organisationName != "") {
```
Comment (bdimitrovski) [inline cypress/models/organisationModule.js:116]: Why do you check if the organization name is empty here? You clear it in the line below nevertheless.

File: cypress/models/registrationModule.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		email = data.user.email, 
+		password = data.user.password,
+		numberOfUsers = data.numberOfUsers.validNumber
+	}) {	
+		this.emailField.clear();
+		this.passwordField.clear();
+		this.numberOfUsers.clear();
+
+		if (email !== "") {
+		this.emailField.should("be.visible").type(email);
```
Comment (bdimitrovski) [inline cypress/models/registrationModule.js]: No need for `.should('be.visible')` assertions, `type()` has got our back.

**Review by bdimitrovski — APPROVED** [review]
Sehr gut! The only thing I don't like is that oldies JS and some assertions that are not really needed.

File: cypress/models/organisationModule.js (hunk 3 above)
Comment (MicmladenTo) [reply cypress/models/organisationModule.js:116]: The issue was in the fact that I cannot merely enter `""`, but I have to skip the field entirely and go straight for the click.
So the idea was to check if the input field is empty. If it is, just click, but if it is not, clear the field and type something.

**Pull request merged**